core/src/main/scala/org/apache/spark/util/collection/PartitionedSerializedPairBuffer.scala

```diff
@@ -72,15 +79,14 @@ private[spark] class PartitionedSerializedPairBuffer[K, V](
       throw new Exception(s"Can't grow buffer beyond ${1 << 31} bytes")
```

Comment:
`1 << 31` is wrong. Maybe it's better to add a method `maxCapacity` to `ChainedBuffer`.